website/util/__init__.py

```diff
@@ -138,14 +140,19 @@ def waterbutler_url_for(route, provider, path, node, user=None, **kwargs):
         url.args['cookie'] = user.get_or_create_cookie()
     elif website_settings.COOKIE_NAME in request.cookies:
         url.args['cookie'] = request.cookies[website_settings.COOKIE_NAME]
+    elif 'Authorization' in request.headers:
+        url.args['token'] = request.headers['Authorization'].replace('Bearer ', '')
```

Comment:
Authorization header could also be `Bearer ...` or `Basic ...` authentication. Might be better off checking `session.data.get('auth_user_access_token')`.